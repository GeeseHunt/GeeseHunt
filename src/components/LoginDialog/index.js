import React from 'react';
import {
  Dialog,
  DialogTitle,
  DialogContent,
  Button,
  DialogActions,
  NoSsr,
} from '@material-ui/core';
import FacebookIcon from '@material-ui/icons/Facebook';
import PropTypes from 'prop-types';

const LoginDialog = ({ open, onClose }) => {
  return (
    <NoSsr>
      <Dialog
        open={open}
        onClose={onClose}
        aria-labelledby="login-dialog-title"
      >
        <DialogTitle id="login-dialog-title">
          Login with our provider
        </DialogTitle>
        <DialogContent>
          <a
            href="/login/facebook"
            style={{ textDecoration: 'none' }}
            onClick={onClose}
          >
            <Button variant="contained" size="large" color="primary">
              <FacebookIcon fontSize="large" />
              &nbsp; Continue with Facebook
            </Button>
          </a>
        </DialogContent>
        <DialogActions />
      </Dialog>
    </NoSsr>
  );
};

LoginDialog.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
};

LoginDialog.defaultProps = {
  open: false,
};

export default LoginDialog;
