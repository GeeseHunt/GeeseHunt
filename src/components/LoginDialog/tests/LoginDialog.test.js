/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import sinon from 'sinon';
import LoginDialog from '..';

describe('LoginDialog', () => {
  const shallow = createShallow();
  let props;

  beforeEach(() => {
    props = { onClose: sinon.spy() };
  });

  afterEach(() => {
    sinon.restore();
  });

  describe('#render', () => {
    it('renders without errors', () => {
      const wrapper = shallow(<LoginDialog {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
