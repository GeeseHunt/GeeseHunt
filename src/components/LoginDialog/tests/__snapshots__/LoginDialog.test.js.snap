// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginDialog #render renders without errors 1`] = `
<NoSsr>
  <WithStyles(ForwardRef(Dialog))
    aria-labelledby="login-dialog-title"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="login-dialog-title"
    >
      Login with our provider
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))>
      <a
        href="/login/facebook"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          size="large"
          variant="contained"
        >
          <FacebookIcon
            fontSize="large"
          />
          Â  Continue with Facebook
        </WithStyles(ForwardRef(Button))>
      </a>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions)) />
  </WithStyles(ForwardRef(Dialog))>
</NoSsr>
`;
